src/core/data.jl

```diff
@@ -1,23 +1,242 @@
+    if _IMs.ismultinetwork(network)
+        Memento.warn(_PMs._LOGGER, "count_repairable_items supports single networks.  Attempting to select network 0.")
+        if haskey(network["nw"],"0")
+            Memento.info(_PMs._LOGGER, "Network 0 found.")
+            nw = network["nw"]["0"]
+        else
+            Memento.error(_PMs._LOGGER, "Network 0 not found.")
+        end
+    else
+        nw = network
+    end
+
+    repairable_count = 0
+    for (comp_name, status_key) in _PMs.pm_component_status
+        for (i, comp) in get(nw, comp_name, Dict())
+            if haskey(comp, status_key) && comp[status_key] != _PMs.pm_component_status_inactive[comp_name] && haskey(comp, "damaged") && comp["damaged"] == 1
+                repairable_count += 1
+            end
+        end
+    end
+    
+    return repairable_count
+end
+
+
+""
+function get_repairable_items(network::Dict{String, Any})
+    ## TODO verify functionality
+    if haskey(network, "multinetwork") && network["multinetwork"] == true
+        repairs = Dict{String,Any}("nw"=>Dict{String,Any}())
+        for (nw, net) in network["nw"]
+            repairs["nw"][nw] = _get_repairable_items(net)
+        end
+    else
+        repairs =  _get_repairable_items(network)
+    end
+    return repairs
+end
+
+
+""
+function _get_repairable_items(network::Dict{String,Any})
+    repairs = Array{Tuple{String,String},1}()
+    for (comp_name, status_key) in _PMs.pm_component_status
+        for (comp_id, comp) in get(network, comp_name, Dict())
+            if haskey(comp, status_key) && comp[status_key] != _PMs.pm_component_status_inactive[comp_name] && haskey(comp, "damaged") && comp["damaged"] == 1
+                push!(repairs, (comp_name, comp_id))
+            end
+        end
+    end
+    return repairs
+end
+
+
+"Count the number of items that have an active status value in a network"
+function count_active_items(network::Dict{String, Any})
+    if _IMs.ismultinetwork(network)
+        network["multinetwork"]
+        Memento.warn(_PMs._LOGGER, "count_active_items supports single networks.  Attempting to select network 0.")
+        if haskey(network["nw"],"0")
+            Memento.info(_PMs._LOGGER, "Network 0 found.")
+            nw = network["nw"]["0"]
+        else
+            Memento.error(_PMs._LOGGER, "Network 0 not found.")
+        end
+    else
+        nw = network
+    end
+
+    active_count = 0
+    for (comp_name, status_key) in _PMs.pm_component_status
+        for (comp_id, comp) in nw[comp_name]
+            if comp[status_key] !==_PMs.pm_component_status_inactive[comp_name]
+                active_count += 1
+            end
+        end
+    end
+        
+    return active_count
+end
+
+
 "Set damage status for damaged_items in nw_data"
-function damaged_items!(nw_data::Dict{String, Any}, damaged_items::Dict{String, Any})
+function damage_items!(nw_data::Dict{String,<:Any}, comp_list::Array{Tuple{String,String},1})
```

Comment:
A better data organization might be `Dict{String, Array{String}}`.  For each component type, a list of component ids.

Reply:
That's a good idea